package/base-files/files/lib/functions/system.sh

```diff
@@ -224,3 +224,15 @@ macaddr_canonicalize() {
 
 	printf "%02x:%02x:%02x:%02x:%02x:%02x" 0x${canon// / 0x} 2>/dev/null
 }
+
+rename_interface_by_mac() {
+	local interface_name="$1"
+	local interface_mac="$2"
+	for device in $(ls -1 /sys/class/net) ; do
+		[ -d "/sys/class/net/$device/device" ] && \
+		[ -f "/sys/class/net/$device/address" ] && \
+		[ "$(cat /sys/class/net/$device/address)" = "$interface_mac" ] && {
+			ip link set "$device" name "$interface_name"
```

Comment:
This will break if multiple devices have the same MAC address?

Reply:
- it ignores 'virtual devices' that have same mac as physic (bridges, vlans and such).
- it works only for <actual device> that have device directory that describe the actual device and that have mac address.
- it expects <actual devices> to have unique mac addresses or function does not work for them

so yes, if let's say person have 4 ethX devices that have the same mac address (i.e. `DE:AD:BE:EF:00:00`) - that function won't work for them

another example - if person has 4 ethX devies that have unique mac addresses (i.e. `DE:AD:BE:EF:00:00`, `DE:AD:BE:EF:00:01`, `DE:AD:BE:EF:00:02` and `DE:AD:BE:EF:00:04`) - that function will work for them.

Reply:
Okay, that's what I assumed. So, since MAC addresses being not unique happens frequently, a function relying on uniqueness should not be added. After all, I really don't think this mechanism will bring us forward.
I'd very much prefer to just hardcode the interface rename without the detour of using MAC addresses.

Reply:
>  So, since MAC addresses being not unique happens frequently, a function relying on uniqueness should not be added.

mac addresses are unique across the octeon target.

> I'd very much prefer to just hardcode the interface rename without the detour of using MAC addresses.

kernel does not guarantee interface naming.
what if order change because of any other reason? that would break logic here.

renaming using mac address imho would be a much more resilient solution for the problem here compared to renaming using interface name since interface names might change in the future but mac addresses board initialize with is known and will not be changed.

Reply:
> mac addresses are unique across the octeon target.

How can you say that? Do you know the MAC address assignment of all devices people use with generic image? Every vendor does it differently ...
The solution could work for your special limited case, but I still don't like to establish such a mechanism which will be used by others then and create problems elsewhere.

> kernel does not guarantee interface naming order.
> what if order change because of any other reason? that would break logic here.

Then we would update and provide a migration script.

Unless order is known to vary with the same exact kernel (i.e. be random intrinsically), I'd prefer to bind to interface names.

You might also split this discussion from the device support itself by using interface names for now, and then add the MAC address-based assignment later in separate patch.

Reply:
>then add the MAC address-based assignment later in separate patch.

Reply:
> You might also split this discussion from the device support itself by using interface names for now, and then add the MAC address-based assignment later in separate patch.
if we

Reply:
> How can you say that? Do you know the MAC address assignment of all devices people use with generic image? Every vendor does it differently ...

every single device octeons tree have (octeon+) - have device tree filled with static board specific mac addresses. that is their ‘feature’.
 there is only few boards in a tree and they are fine to use rename by mac since their mac addresses are mapped in device tree already.

if there is no generic board currently that don’t have mac set.

and yeah, we are not forcing people to use that rename feature.
if board fit the requirements - that rename feature could be used. if it is not - it would not. it is as simple as that.

> Then we would update and provide a migration script.

it sounds like ‘create a problem and provide a solution’ to me instead of providing a resilient solution.

i disagree about using device names to rename because that solution will be very fragile.

we better not rename devices at'all at this point of time if renaming by mac creates so much trouble.

i will update a MR removing rename feature completely and we will stick with whatever kernel provides.

Reply:
so what should we do to interface rename?
i am okay with removing that feature altogether (instead of adding a band aid to rename by interface name) if that is gonna make merge happen.